--- fpcbuild-2.6.4/fpcsrc/packages/gdbint/src/gdbint.pp~	2016-04-01 17:53:51.000000000 +0900
+++ fpcbuild-2.6.4/fpcsrc/packages/gdbint/src/gdbint.pp	2016-04-01 23:08:14.978738263 +0900
@@ -59,6 +59,17 @@
 
 {$undef GDB_VERSION_RECOGNIZED}
 
+{ 7.10.x }
+{$ifdef GDB_V710}
+  {$info using gdb 7.10.x}
+  {$define GDB_VERSION_RECOGNIZED}
+  {$define GDB_VER_GE_710}
+{$endif}
+
+{$ifdef GDB_VER_GE_710}
+  {$define GDB_VER_GE_709}
+{$endif}
+
 { 7.9.x }
 {$ifdef GDB_V709}
   {$info using gdb 7.9.x}
