--- fpcbuild-3.0.0/install/man/Makefile.orig	2011-06-19 20:40:37.000000000 +0200
+++ fpcbuild-3.0.0/install/man/Makefile	2016-12-17 20:15:03.522162624 +0100
@@ -31,8 +31,6 @@
 	$(GINSTALL) -d -m 755 $(INSTALL_MANDIR)/man5
 	$(GINSTALL) -c -m 644 man1/*.1 $(INSTALL_MANDIR)/man1
 	$(GINSTALL) -c -m 644 man5/*.5 $(INSTALL_MANDIR)/man5
-	find man1/*.1 -exec gzip -9f $(INSTALL_MANDIR)/'{}' \;
-	find man5/*.5 -exec gzip -9f $(INSTALL_MANDIR)/'{}' \;
 ifdef NOIDE
 	rm $(INSTALL_MANDIR)/man1/fp.*
 endif
