Index: fpcdocs/Makefile
===================================================================
--- fpcdocs/Makefile	(revision 1259)
+++ fpcdocs/Makefile	(working copy)
@@ -2672,9 +2672,9 @@
 	$(RTLMAKESKEL) $(RTLCHARACTER) --output=character.new.xml
 	./cleanxml $(RTLNEWXML)
 rtl.inc: $(RTLXML)
-	$(FPDOC) --output=rtl.inc --project=rtl-project.xml --format=latex
+	$(FPDOC) --base-input-dir=$(FPCSRCDIR)/rtl --output=rtl.inc --project=rtl-project.xml --format=latex
 fcl.inc: $(FCLXML)
-	$(FPDOC) --output=fcl.inc --project=fcl-project.xml --format=latex
+	$(FPDOC) --base-input-dir=$(FPCSRCDIR)/packages --output=fcl.inc --project=fcl-project.xml --format=latex
 fclres.inc: $(FCLRESXML)
 	$(FPDOC) --output=fclres.inc $(FCLRESOPTS) --format=latex
 RTFFILES = $(addsuffix .rtf,$(RTFS))
Index: fpcdocs/Makefile.fpc
===================================================================
--- fpcdocs/Makefile.fpc	(revision 1258)
+++ fpcdocs/Makefile.fpc	(working copy)
@@ -777,10 +777,10 @@
 #####################################################################
 
 rtl.inc: $(RTLXML)
-        $(FPDOC) --output=rtl.inc --project=rtl-project.xml --format=latex
+        $(FPDOC) --base-input-dir=$(FPCSRCDIR)/rtl --output=rtl.inc --project=rtl-project.xml --format=latex
 
 fcl.inc: $(FCLXML)
-        $(FPDOC) --output=fcl.inc --project=fcl-project.xml --format=latex
+        $(FPDOC) --base-input-dir=$(FPCSRCDIR)/packages --output=fcl.inc --project=fcl-project.xml --format=latex
 
 fclres.inc: $(FCLRESXML)
         $(FPDOC) --output=fclres.inc $(FCLRESOPTS) --format=latex
